test/unit/kubernetes-deploy/google_config_test.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class GoogleConfigTest < KubernetesDeploy::TestCase
+  def setup
+    WebMock.disable_net_connect!
+  end
+
+  def teardown
+    WebMock.allow_net_connect!
+  end
+
+  def test_auth_use_default_gcp_success
```

Comment:
I think it also makes sense to check that the config works as usual if auth provider isn't GCP.